diff --git mlibc-clean/meson.build mlibc-workdir/meson.build
index bf45f16..222509b 100644
--- mlibc-clean/meson.build
+++ mlibc-workdir/meson.build
@@ -320,19 +320,6 @@ subdir('options/bsd')
 rtlib_deps = []
 
 if not headers_only
-	libgcc = meson.get_compiler('c').find_library('gcc', required: false)
-
-	compiler_rt_name = 'libclang_rt.builtins-' + host_machine.cpu_family()
-	compiler_rt = meson.get_compiler('c').find_library(compiler_rt_name, required: false)
-
-	if libgcc.found()
-		rtlib_deps += libgcc
-	elif compiler_rt.found()
-		rtlib_deps += compiler_rt
-	else
-		error('neither libgcc nor ' + compiler_rt_name + ' was found')
-	endif
-
 	ld_cpp_args = [
 		'-fvisibility=hidden',
 		'-fno-stack-protector',
